### PR TITLE
[14.0][IMP] Added the field 'color' to partner_id_category.

### PR DESCRIPTION
**Pull request by Chanakya-OSI**

### CODE REVIEW

**Comment by patrickrwilson** [comment]
@Chanakya-OSI Please fix pre-commit then this should be ready for review. Thanks for adding this.

**Review by patrickrwilson — APPROVED** [review]
LGTM

**Comment by patrickrwilson** [comment]
@rousseldenis do you mind reviewing this?

**Review by ajaniszewska-dev — APPROVED** [review]

**Comment by NL66278** [comment]
@Chanakya-OSI How and where is this color used? I see no way to set or to display it...

File: partner_identification/models/res_partner_id_category.py [hunk 1]
```diff
@@ -18,6 +20,10 @@ class ResPartnerIdCategory(models.Model):
     _description = "Partner ID Category"
     _order = "name"
 
+    def _get_default_color(self):
+        return randint(1, 11)
+
+    color = fields.Integer(string="Color Index", default=_get_default_color)
```
Comment (rousseldenis) [inline partner_identification/models/res_partner_id_category.py:26]: Better to use a lambda for default in order to allow override.

Comment (dreispt) [reply partner_identification/models/res_partner_id_category.py:26]: IMO Better use a method to allow extending.

**Comment by dreispt** [comment]
@Chanakya-OSI This PR is blocked because it is expecting ans answer from you. Thank you.

**Comment by Chanakya-OSI** [comment]
> @Chanakya-OSI How and where is this color used? I see no way to set or to display it...



> @Chanakya-OSI How and where is this color used? I see no way to set or to display it...

This color field is used in the modules 'partner_identification_sale' as well as 'partner_identification_stock'.

File: partner_identification/models/res_partner_id_category.py (hunk 1 above)
Comment (Chanakya-OSI) [reply partner_identification/models/res_partner_id_category.py:26]: Yes keeping method for allowing extending.

**Pull request merged**